util/llbutil/secretprovider/cloud.go

```diff
@@ -23,12 +26,63 @@ func NewCloudStore(client cloud.Client) secrets.SecretStore {
 // secrets are referenced via +secret/name or +secret/org/name (or +secret/org/subdir1/.../name)
 // however by the time GetSecret is called, the "+secret" prefix is removed.
 func (cs *cloudStore) GetSecret(ctx context.Context, id string) ([]byte, error) {
-	if !strings.HasPrefix(id, "/") {
-		return nil, secrets.ErrNotFound
-	}
-	dt, err := cs.client.Get(ctx, id)
+
+	q, err := url.ParseQuery(id)
 	if err != nil {
-		return nil, err
+		return nil, errors.New("failed to parse secret ID")
+	}
+
+	var data []byte
+
+	name := q.Get("name")
+	if name == "" {
+		return nil, errors.New("name parameter not found")
+	}
+
+	switch q.Get("v") {
+	case "0": // Legacy secret ID format includes the name only
+
+		// This format requires the secret to be prefixed with either <org-name>
+		// or 'user'.
+		if !strings.Contains(name, "/") {
+			return nil, secrets.ErrNotFound
+		}
```

Comment:
I think that this is not right. I think we allowed `/` to not be present for local-only secrets. I could be wrong though? @alexcb to comment?

Reply:
This code is used to check whether secrets exist in the cloud repository. It shouldn't impact local secrets. Old-style secret names need to be formatted like `/<org>/<secret` if I'm not mistaken.

Reply:
Oh ok - yeah, I think we simply use a completely separate secret provider for the local secrets, right?

Conversely, would the user be able to override a cloud secret via a locally passed in secret (without crafting this query param format on their own) in the new implementation?

Reply:
Yep, it's in `map_store.go`.

Yes, definitely. This change should support all existing behaviors around local/env secrets. The only real difference is that one needs to use `PROJECT` when using the new project-based cloud secrets option.

Reply:
I believe. That was my intention. I'll add some tests for that case.

Reply:
> I think that this is not right. I think we allowed `/` to not be present for local-only secrets. I could be wrong though? @alexcb to comment?

This check is expected here (as Mike pointed out this code already existed).